rotkehlchen/assets/utils.py

```diff
@@ -144,6 +146,12 @@ def get_or_create_evm_token(
     - NotERC721Conformant exception if an ethereum manager is given to query
     and the given address does not conform to ERC721 spec
     """
+    if chain_id is None:
+        assert evm_inquirer is not None, 'evm_inquirer must be provided if chain_id is not'
+        chain_id = evm_inquirer.chain_id
+    else:
+        assert evm_inquirer is None, 'evm_inquirer must not be provided if chain_id is'
+
```

Comment:
Remove this. We do not do dynamic argument checking for things that depend solely on us. You can achieve stuff like that via types but still not needed, is kind of an overkill.

And also why do this at all? Since this depends solely on us, but provide chain id. Be disciplined.

Reply:
Alright